src/Lykke.Job.CandlesHistoryWriter.Services/Candles/CandlesCacheInitalizationService.cs

```diff
@@ -116,7 +116,12 @@ private async Task CacheAssetPairCandlesAsync(AssetPair assetPair, DateTime now,
                 {
                     foreach (var timeInterval in Constants.StoredIntervals)
                     {
-                        var candles = await _candlesHistoryRepository.GetExactCandlesAsync(assetPair.Id, timeInterval, priceType, now, _amountOfCandlesToStore);
+                        var candles = (await _candlesHistoryRepository.GetExactCandlesAsync(assetPair.Id, timeInterval, priceType, now, _amountOfCandlesToStore)).ToArray();
```

Comment:
Change return type of the ```GetExactCandlesAsync``` to the ```Task<IReadOnlyCollection<ICandle>>``` this will let to remove extra ```ToArray()``` call here.